v2/federation/watch_test.go

```diff
@@ -0,0 +1,103 @@
+package federation
```

Comment:
I don't see a test case that tests getting more than one update (i.e. that the watch loop is functioning as advertised)